flepimop/gempyor_pkg/src/gempyor/sync/_sync.py

```diff
@@ -455,26 +496,17 @@ def _sync_pydantic(
         if not self.sync:
             logger.info("No protocols to sync.")
             return run(["echo", "No protocols to sync"], check=True)
-        target_protocol = (
-            sync_options.protocol if sync_options.protocol else list(self.sync.keys())[0]
-        )
-        logger.debug("Resolved protocol: %s.", str(target_protocol))
-        if proto := self.sync.get(target_protocol):
-            logger.info("Executing protocol: %s.", str(proto))
-            return proto.execute(sync_options, verbosity)
-        logger.error(
-            "Protocol '%s' not found, available protocols are: %s.",
-            target_protocol,
-            ", ".join(self.sync.keys()),
-        )
-        return run(
-            [
-                "echo",
-                f"Protocol '{target_protocol}' not found, available "
-                f"protocols are: {', '.join(self.sync.keys())}.",
-            ],
-            check=True,
-        )
+        if (sync_model := _get_sync_protocol(self.sync, sync_options.protocol)) is None:
+            return run(
+                [
+                    "echo",
+                    f"Protocol '{sync_options.protocol}' not found, available "
+                    f"protocols are: {', '.join(self.sync.keys())}.",
+                ],
+                check=True,
+            )
```

Comment:
seems like _get_sync_protocol throws under this condition; should this be a try-catch style check? or change _get_sync_protocol to not raise anything and return `SyncModel | None`?

I'm a ambivalent about which to do